src/api/views.py

```diff
@@ -496,4 +513,161 @@ def check_license(request):
         else:
             return Response(
                 serializer.errors, status=status.HTTP_400_BAD_REQUEST
-                )        
\ No newline at end of file
+                )
+
+
+@api_view(['GET', 'POST'])
+@renderer_classes((JSONRenderer,))
+@permission_classes((AllowAny, ))
+def submit_license(request):
+    """ Handle submit license api request """
+    if request.method == 'GET':
+        """ Return all check license api request """
+        query = SubmitLicenseModel.objects.all()
+        serializer = SubmitLicenseSerializer(query, many=True)
+        return Response(serializer.data)
+
+    elif request.method == 'POST':
+        """ Return the result of license submittal on the post license details """
+        githubClientId = settings.SOCIAL_AUTH_GITHUB_KEY
+        githubClientSecret = settings.SOCIAL_AUTH_GITHUB_SECRET
+        #state = ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(16))
+        url = 'https://github.com/login/oauth/authorize/?client_id={}'.format(githubClientId)
+        browser = webdriver.Firefox()
```

Comment:
> * The application that is calling the API generates the code and passes that in as a parameter in the POST

This was the first ever approach I tried but discarded it as I thought the user will have to pass the authentication code manually. Nevertheless I think this approach would suit the best.

> To make it easy to test, we could move the webdriver code to a separate method or class to allow an interactive test of the license submittal API. My preference would be a separate class to keep the dependencies in this class to a minimum

If we are asking the user for the authentication code then I don't see any need for using the selenium webdriver which will also make the process of testing way easier.

@goneall I liked your approach, just let me know if you want me to implement it and I will do it right now and update the branch.

Reply:
@Ugtan Please go ahead and update it with this approach.  The user can always implement a browser approach to gather the username/password as part of the protocol.  We just need to document this well so that the caller knows what the protocol is.